lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -268,6 +273,11 @@ export function WithRpcSessions({ children }: { children: React.ReactNode }) {
         []
     )
 
+    useEvent(ec.events.serverRestarted, () => {
+        sessions.dispose();
+        setSessions(new RpcSessions(ec));
```

Comment:
How does this deal with the possibility that there could be some pending async calls to `sendClientRequest` or `await sesh.call` that could complete after the new RpcSessions has been created, then you could have competing async requests, etc...  I'd feel better if RpcSessions had a this.disposed flag that each async completion checked and bailed if it is set to true.

Reply:
Good point.  A lot of this code assumes that the `#connected` and `#connecting` maps stay the same (even though the RpcSessions object is recreated every time a session is created).  I don't want to refactor this now.